### PR TITLE
ECF-20: Add a deploy action for develop branch

### PR DESCRIPTION
**Pull request by kicferk1**
### Context
We want some automatic deployment to PaaS

### Changes proposed in this pull request
Adds a github action that will deploy develop branch to PaaS on push

### Guidance to review
Largely stolen from teaching-vacancies

And simplified along the way.

We will likely store secrets differently. But this is a working reference point for future work.

### CODE REVIEW

**Review by amazimbe — APPROVED** [review]
Looks good to me

**Comment by kicferk1** [comment]
This is a work in progress - I will add a bunch of secrets to github tomorrow, and eventually remove this branch from the list of branches we run this action on :)

File: .github/workflows/deploy.yml [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+    needs: turnstyle
+    runs-on: ubuntu-20.04
+    steps:
+      - uses: actions/checkout@v2
+        name: Checkout Code
+
+      - name: Pin Terraform version
+        uses: hashicorp/setup-terraform@v1.2.1
+        with:
+          terraform_version: 0.13.4
```
Comment (ethanmills) [inline .github/workflows/deploy.yml:38]: Let's use 0.14.0 if we can

File: .github/workflows/deploy.yml [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+
+      - name: Set environment variables
+        id: set_env_vars
+        run: |
+          echo "TAG=${{ github.sha }}" >> $GITHUB_ENV
+
+      - name: Set up Ruby
```
Comment (ethanmills) [inline .github/workflows/deploy.yml:45]: Do we need this?

File: .github/workflows/deploy.yml [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+        name: Checkout Code
+
+      - name: Pin Terraform version
+        uses: hashicorp/setup-terraform@v1.2.1
+        with:
+          terraform_version: 0.13.4
+
+      - name: Set environment variables
+        id: set_env_vars
+        run: |
```
Comment (ethanmills) [inline .github/workflows/deploy.yml:42]: What does the pipe do here?

File: .github/workflows/deploy.yml [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+      - name: Deploy to dev
+        env:
+          AWS_ACCESS_KEY_ID: ${{ secrets.AWS_ACCESS_KEY_ID }}
+          AWS_SECRET_ACCESS_KEY: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+          TF_VAR_paas_user: ${{ secrets.GOVPAAS_DEV_USERNAME }}
+          TF_VAR_paas_password: ${{ secrets.GOVPAAS_DEV_PASSWORD }}
+        run: |
+          export TF_VAR_paas_app_docker_image=${{ env.DOCKERHUB_REPOSITORY }}:develop
+          cd terraform/app
+          terraform init -reconfigure -input=false -backend-config="bucket=paas-s3-broker-prod-lon-7f2ca242-9929-4662-a79c-c454ea56ea7b"
```
Comment (ethanmills) [inline .github/workflows/deploy.yml:75]: The bucket will need extracting when we do staging, but fine for now

File: Dockerfile [hunk 5]
```diff
@@ -4,8 +4,8 @@ ARG BASE_RUBY_IMAGE=ruby:2.7.1-alpine
 # building all layers above it if a value is not specidied during the build
 ARG BASE_RUBY_IMAGE_WITH_GEMS_AND_NODE_MODULES=early-careers-framework-gems-node-modules
 
-# Stage 1: install-gems-node-modules, download gems and node modules.
-FROM ${BASE_RUBY_IMAGE} AS install-gems-node-modules
+# Stage 1: Dwnload gems and node modules.
```
Comment (ethanmills) [inline Dockerfile:7]: typo

**Review by ethanmills — APPROVED** [review]
One typo in a comment, looks good!

**Pull request merged**